nucypher/characters/control/interfaces.py

```diff
@@ -220,6 +221,19 @@ def retrieve(self,
 
         self.character.join_policy(label=label, publisher_verifying_key=alice_verifying_key)
 
+        if self.character.federated_only:
+            from nucypher.policy.maps import TreasureMap as _MapClass
+        else:
+            from nucypher.policy.maps import SignedTreasureMap as _MapClass
+
+        # TODO: This LBYL is ugly and fraught with danger.  NRN
+        if isinstance(treasure_map, bytes):
+            treasure_map = _MapClass.from_bytes(treasure_map)
+
+        if isinstance(treasure_map, str):
+            tmap_bytes = treasure_map.encode()
+            treasure_map = _MapClass.from_bytes(b64decode(tmap_bytes))
+
```

Comment:
Ideally, this would be handled as part of the "TreasureMap" field deserialization. In Porter, I do something similar for validation, but I only use the splitter since I only care about the bytes representation that I then pass on to ursula(s) - https://github.com/nucypher/nucypher/pull/2664/files#diff-54f7a125485aa11f633996b09e5f6b0468b8b91cd06a04cb097a808b70e14c1eR59.

However, we can think about changing the field to always make the `_deserialize()` function return a TreasureMap/SignedTreasureMap object (instead of bytes) and then use the object directly in this case, or convert the object back to bytes for usage for Porter.

Reply:
The problem here is that in the current form it needs to know if the character is federated, and this is information is not available in `Field._deserialize()`. We really need to try and get rid of this dependency.

Reply:
This was the motivation behind the following in the Porter EPIC, https://github.com/nucypher/nucypher/pull/2664/commits/1eda693733bcefbc970786e63afd366901d5a33c. You know the nature of the character (federated vs non-federated) when the interface is initiated so you can set a schema context.

Perhaps as a follow-up PR that already includes Porter changes.

Reply:
The best solution in the long-term is to unify federated and unfederated treasure maps #2737 but this will require some additional footwork.

Reply:
Filed #2751 as a pretty easy interim solution until #2737 can actually be done.